Solutions/Marain.UserNotifications.Management.Host/Marain/UserNotifications/Management/Host/OpenApi/GenerateTemplateService.cs

```diff
@@ -0,0 +1,180 @@
+
+            // Get the user preference for the userId
+            UserPreference? userPreference = await userPreferencesStore.GetAsync(body.UserIds[0]).ConfigureAwait(false);
+
+            // Check if the user has set the communication channels for the incoming notification type
+            if (userPreference == null)
+            {
+                throw new UserNotificationNotFoundException($"There is no user preference set up for this user {body.UserIds[0]} for tenant {tenant.Id}");
+            }
+
+            if (userPreference.CommunicationChannelsPerNotificationConfiguration == null)
+            {
+                throw new UserNotificationNotFoundException($"There are no communication channel set up for the user {body.UserIds[0]} for tenant {tenant.Id}");
+            }
+
+            if (!userPreference.CommunicationChannelsPerNotificationConfiguration.ContainsKey(body.NotificationType))
+            {
+                throw new UserNotificationNotFoundException($"There is no communication channel set up for the user {body.UserIds[0]} for notification type {body.NotificationType} for tenant {tenant.Id}");
+            }
+
+            List<CommunicationType>? registeredCommunicationChannels = userPreference.CommunicationChannelsPerNotificationConfiguration[body.NotificationType];
+
+            if (registeredCommunicationChannels == null || registeredCommunicationChannels.Count == 0)
+            {
+                throw new Exception($"There are no communication channel set up for the user {body.UserIds[0]} for notification type {body.NotificationType} for tenant {tenant.Id}");
+            }
+
+            // Gets the AzureBlobTemplateStore
+            INotificationTemplateStore templateStore = await this.tenantedTemplateStoreFactory.GetTemplateStoreForTenantAsync(tenant).ConfigureAwait(false);
+
+            // Get the notification template for the notification type
+            NotificationTemplate? rawNotificationTypeTemplate = await templateStore.GetAsync(body.NotificationType).ConfigureAwait(false);
+
+            if (rawNotificationTypeTemplate == null)
+            {
+                throw new UserNotificationNotFoundException($"There is no notification template set up for the user {body.UserIds[0]} for notification type {body.NotificationType} for tenant {tenant.Id}");
+            }
+
+            var communicationTypeDeliveryStatus = new Dictionary<CommunicationType, bool>();
+            EmailTemplate? emailTemplate = null;
+            SmsTemplate? smsTemplate = null;
+            WebPushTemplate? webPushTemplate = null;
+            Dictionary<string, object> existingProperties = PropertyBagHelpers.GetDictionaryFromPropertyBag(body.Properties);
+
+            foreach (CommunicationType channel in registeredCommunicationChannels)
+            {
+                switch (channel)
+                {
+                    case CommunicationType.Email:
+
```

Comment:
Extra line space; I'm surprised this wasn't caught by StyleCop?

Reply:
Yeah, strange. Done.